sway-features/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+            ];
+
+            #[derive(Copy, Clone, Debug, ValueEnum)]
+            #[value(rename_all = "snake")]
+            pub enum Features {
+                $(
+                    [<$name:camel>],
+                )*
+            }
+
+            impl std::str::FromStr for Features {
+                type Err = Error;
+
+                fn from_str(s: &str) -> Result<Self, Self::Err> {
+                    match s {
+                        $(
+                            stringify!([<$name:snake>]) => {
+                                Ok(Self::[<$name:camel>])
+                            },
+                        )*
+                        _ => Err(Error::UnknownFeature(s.to_string())),
+                    }
+                }
+            }
+
+            #[derive(Copy, Clone, Debug, PartialEq, Eq)]
+            pub struct ExperimentalFeatures {
+                $(
+                    pub [<$name:snake>]: bool,
+                )*
+            }
+
+            impl std::default::Default for ExperimentalFeatures {
+                fn default() -> Self {
+                    Self {
+                        $(
+                            [<$name:snake>]: $enabled,
+                        )*
+                    }
+                }
+            }
+
+            impl ExperimentalFeatures {
+                pub fn set_enabled(&mut self, feature: &str, enabled: bool) -> Result<(), Error> {
+                    let feature = feature.trim();
+                    match feature {
+                        $(
+                            stringify!([<$name:snake>]) => {
+                                self.[<$name:snake>] = enabled;
+                                Ok(())
+                            },
+                        )*
+                        "" => Ok(()),
+                        _ => Err(Error::UnknownFeature(feature.to_string())),
+                    }
+                }
+
+                pub fn enable_feature(&mut self, feature: Features, enabled: bool) {
+                    match feature {
+                        $(
+                            Features::[<$name:camel>] => {
+                                self.[<$name:snake>] = enabled
+                            },
+                        )*
+                    }
+                }
+
+                pub fn is_enabled_by_cfg(&self, cfg: &str) -> Result<bool, Error> {
+                    match cfg {
+                        $(
+                            stringify!([<experimental_ $name:snake>]) => Ok(self.[<$name:snake>]),
+                        )*
+                        _ => Err(Error::UnknownFeature(cfg.to_string()))
+                    }
+                }
+
+                $(
+                pub fn [<with_ $name:snake>](mut self, enabled: bool) -> Self {
+                    self.[<$name:snake>] = enabled;
+                    self
+                }
+                )*
+            }
+
+            // enum ExperimentalFeatureError {
```

Comment:
Should we have this enum? If not, let's rather remove the commented code.

Reply:
Done.